index.bs

```diff
@@ -8972,6 +9018,371 @@ worker=] algorithm:
 
 </div>
 
+## The storage Module ## {#module-storage}
+
+The <dfn export for=modules>storage</dfn> module contains functionality and
+events related to storage.
+
+A <dfn>storage partition</dfn> is a namespace within which the user agent may
+organize persistent data such as [=cookies=] and local storage.
+
+A <dfn>storage partition key</dfn> uniquely identifies a [=storage partition=].
+
+### Definition ### {#module-storage-definition}
+
+[=Remote end definition=]
+
+<pre class="cddl remote-cddl">
+StorageCommand = (
+  storage.GetCookies //
+  storage.SetCookie //
+  storage.DeleteCookies
+)
+</pre>
+
+[=Local end definition=]
+
+<pre class="cddl local-cddl">
+StorageResult = (
+  storage.GetCookiesResult //
+  storage.SetCookieResult //
+  storage.DeleteCookiesResult
+)
+</pre>
+
+### Types ### {#module-storage-types}
+
+#### The storage.PartitionKey Type #### {#type-storage-PartitionKey}
+
+[=Remote end definition=] and [=Local end definition=]
+
+<pre class="cddl local-cddl remote-cddl">
+storage.PartitionKey {
+    ? userContext: text,
+    ? sourceOrigin: text,
+    Extensible,
+}
+</pre>
+
+The <code>storage.PartitionKey</code> type represents a [=storage partition key=].
+
+A [=remote end=] has a [=/set=] of <dfn>storage partition key attributes</dfn> which contains zero or more implementation-defined string values.
+
+A [=remote end=] has a [=/map=] of <dfn>default values for storage partition key attributes</dfn> which contains zero or more implementation-defined entries. Each key must be a member of the [=storage partition key attributes=] which the [=remote end=] does not require as input to [=storage.getCookies=], [=storage.setCookie=], and [=storage.deleteCookies=]; the corresponding value is a the value which the [=remote end=] uses when one is not provided.
+
+<div algorithm>
+To <dfn>expand a storage partition spec</dfn> given |partition spec| and |cookie domain|:
+
+1. If |partition spec| is a [=browsing context id=]:
+
+  1. Let |browsing context| be the result of [=trying=] to [=get a browsing context=] given |partition spec|.
+
+  1. Let |partition key| be the [=storage partition key|key=] of |browsing context|'s [=associated storage partition=].
+
+  1. Return [=success=] with data |partition key|.
+
+1. Otherwise, if |partition spec| is null:
```

Comment:
Due to the early return, this step could be expressed with an independent "If" (that is, by removing the word "otherwise"). I'm calling this out because I know opinions differ on which style is preferable.